pkg/products/marin3r/reconciler.go

```diff
@@ -537,3 +524,50 @@ func (r *Reconciler) reconcileServiceMonitor(ctx context.Context, client k8sclie
 
 	return integreatlyv1alpha1.PhaseCompleted, nil
 }
+
+func reconcileEnvoyConfigRevisionsDeletion(ctx context.Context, client k8sclient.Client, installation integreatlyv1alpha1.RHMI) (integreatlyv1alpha1.StatusPhase, error) {
+	listOptions := []k8sclient.ListOption{
+		k8sclient.MatchingLabels(map[string]string{
+			"marin3r.3scale.net/envoy-api": "v2",
+		}),
+		k8sclient.InNamespace(fmt.Sprintf("%s3scale", installation.Spec.NamespacePrefix)),
+	}
+
+	envoyConfigRevisions := &v1alpha1.EnvoyConfigRevisionList{}
+	err := client.List(ctx, envoyConfigRevisions, listOptions...)
+	if err != nil {
+		return integreatlyv1alpha1.PhaseFailed, err
+	}
+
+	for _, envoyConfigRevision := range envoyConfigRevisions.Items {
+		// Attempt to delete envoyConfigRevision if there's no deletion timestamp present
+		if envoyConfigRevision.DeletionTimestamp == nil {
+			err := client.Delete(ctx, &envoyConfigRevision)
+			if err != nil {
+				return integreatlyv1alpha1.PhaseFailed, err
+			}
+		}
+
+		// Fetch object again before removing finalizer to ensure that configRevision has not been deleted already by client.Delete call above - this can happen if for any reason
+		// there was no Finalizer on envoyConfigRevision
+		err = client.Get(ctx, k8sclient.ObjectKey{Name: envoyConfigRevision.Name, Namespace: envoyConfigRevision.Namespace}, &v1alpha1.EnvoyConfigRevision{})
+		if k8serr.IsNotFound(err) {
+			// if configRevision is not found, we are good to go
+			continue
+		} else if err != nil {
+			return integreatlyv1alpha1.PhaseFailed, err
+		}
```

Comment:
```suggestion
		if err != nil {
			if k8serr.IsNotFound(err) {
			     // if configRevision is not found, we are good to go
			     continue
			}
			return integreatlyv1alpha1.PhaseFailed, err
		}
```
slightly cleaner without the else. and there's possibility the err in the first check is nil.

Reply:
Thanks. Done!